node/resolvers/directives/withPermissions.ts

```diff
@@ -4,6 +4,15 @@ import type { GraphQLField } from 'graphql'
 import { defaultFieldResolver } from 'graphql'
 import { SchemaDirectiveVisitor } from 'graphql-tools'
 
+import type StorefrontPermissions from '../../clients/storefrontPermissions'
+
+export const getCheckUserPermission = async (
+  storefrontPermissions: StorefrontPermissions
+) =>
+  storefrontPermissions.checkUserPermission().then((result: any) => {
+    return result?.data?.checkUserPermission ?? null
+  })
```

Comment:
At VTEX we prefer using async/await instead of promises (meaning `.then()`, `.catch()`, etc). In the end they're the same thing, as async/await uses promises behind the scenes, but async/await has better readability, so that's why it's preferred (outside VTEX as well).

You can make any function using promises use async/await instead, by replacing `.then` with `await` and `.catch` with a regular try/catch block. This is what it would look like here:

```suggestion
export const getCheckUserPermission = async (
  storefrontPermissions: StorefrontPermissions
) =>
  const result = await storefrontPermissions.checkUserPermission()
  return result?.data?.checkUserPermission ?? null
```

Behind the scenes, the function above will be compiled into a very similar version to what you have in this PR (using promises). It's sugar syntax, same behavior.